### PR TITLE
Loading windows with vuex

### LINKED ISSUE #1006: Refactor UI Design
Placeholder. Will be added ...

### LINKED ISSUE #1037: Error opening window
> Hey, we are a new design for window very good. The change already exists in window-redefinition Branch

Hi Yamel , I checkout window-redefinition branch now , but with it I am not able to see any screen. Always have spinner shown: 
image

_Originally posted by @RUser1990 in https://github.com/adempiere/adempiere-vue/issues/1006#issuecomment-887264480_

### PR DESCRIPTION
**Pull request by elsiosanchez**
#### Screenshot or Gif

Peek 22-07-2021 09-55

#### Other relevant information
- Your OS: Debian 9
- Web Browser: CHrome 
- Node.js version: 18.2.1

#### Issues
Fixes #1006 
FIxes #1037

### CODE REVIEW

File: src/views/ADempiere/Window/windowUtils.js [hunk 1]
```diff
@@ -14,25 +14,19 @@
...
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-import { convertWindow } from '@/utils/ADempiere/apiConverts/dictionary.js'
-
-export function generateWindow(windowResponse) {
-  const responseWindow = convertWindow(windowResponse)
-
+export function generateWindow(responseWindow) {
   const {
-    tabsList, tabsListParent, tabsListChild,
```
Comment (EdwinBetanc0urt) [inline src/views/ADempiere/Window/windowUtils.js:23]: Keep the list of child tabs, this attribute is mandatory to render child tabs.

File: src/views/ADempiere/Window/windowUtils.js [hunk 2]
```diff
@@ -14,25 +14,19 @@
...
     tabs: responseWindow.tabs,
     parentUuid: responseWindow.uuid
   })
-
   const newWindow = {
     ...responseWindow,
     tabsList,
     currentTab: tabsListParent[0],
     tabsListParent,
-    tabsListChild,
```
Comment (EdwinBetanc0urt) [inline src/views/ADempiere/Window/windowUtils.js:35]: Keep the list of child tabs, this attribute is mandatory to render child tabs.

File: src/views/ADempiere/Window/windowUtils.js [hunk 3]
```diff
@@ -86,22 +80,10 @@ export function generateTabs({
       tabParentIndex
     }
   })
-
-  // generate tabs childs
-  const tabsListChild = tabsList.filter(tabItem => {
```
Comment (EdwinBetanc0urt) [inline src/views/ADempiere/Window/windowUtils.js:91]: this is used to render the child tabs in
https://github.com/adempiere/adempiere-vue/blob/window-redefinition/src/views/ADempiere/Window/StandardWindow.vue#L31

**Review by EdwinBetanc0urt — CHANGES_REQUESTED** [review]

File: src/views/ADempiere/Window/index.vue [hunk 4]
```diff
@@ -49,7 +48,8 @@ import { defineComponent, computed, ref } from '@vue/composition-api'
 
 import ActionMenu from '@/components/ADempiere/ActionMenu'
 import LoadingView from '@/components/ADempiere/LoadingView'
-import { generateWindow as generateWindowRespose } from './windowUtils'
+import { convertWindow } from '@/utils/ADempiere/apiConverts/dictionary.js'
```
Comment (yamelsenih) [inline src/views/ADempiere/Window/index.vue:51]: Very old approach. Please keep it as it now

File: src/store/modules/ADempiere/dictionary/window/actions.js [hunk 5]
```diff
@@ -17,12 +17,24 @@
 import { requestWindowMetadata } from '@/api/ADempiere/dictionary/window'
 
 export default {
-  getWindowDefinitionFromServer({ commit }, uuid) {
-    return requestWindowMetadata({
-      uuid
+  addWindow({ commit }, windowResponse) {
```
Comment (yamelsenih) [inline src/store/modules/ADempiere/dictionary/window/actions.js:20]: Already is handle it as mutation

**Review by yamelsenih — CHANGES_REQUESTED** [review]

**Pull request merged**